src/invoker/src/handler.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{
+    api::{InvokeRequest, InvokeResponse, Output},
+    executor::{Executor, SandboxGlobalSettings},
+    graph_interp::Interpreter,
+    print_invoke_request::PrintWrapper,
+};
+use anyhow::Context as _;
+use minion::erased::Backend;
+use std::path::PathBuf;
+
+pub struct HandlerConfig {
+    pub work_dir: PathBuf,
+}
+
+pub struct Handler {
+    cfg: HandlerConfig,
+    sandbox_global_settings: SandboxGlobalSettings,
+    minion_backend: Box<dyn Backend>,
+}
+
+fn setup_minion() -> anyhow::Result<Box<dyn Backend>> {
+    let mut settings = minion::linux::Settings::new();
+    settings.cgroup_prefix = "/jjs".into();
+    let backend = minion::LinuxBackend::new(settings)?;
+    Ok(Box::new(backend))
+}
+
+impl Handler {
+    pub async fn new(
+        config: HandlerConfig,
+        sandbox_global_settings: SandboxGlobalSettings,
+    ) -> anyhow::Result<Self> {
+        let backend = setup_minion().context("failed to initialize minion backend")?;
+
+        Ok(Handler {
+            cfg: config,
+            sandbox_global_settings,
+            minion_backend: backend,
+        })
+    }
+
+    fn print_request(&self, req: &InvokeRequest) {
+        let wrapper = PrintWrapper(req);
+        let msg = wrapper.print();
+        tracing::info!(request = msg.as_str(), "processing InvokeRequest");
+    }
+
+    #[tracing::instrument(skip(self, req))]
+    pub async fn handle_invoke_request(
+        &self,
+        req: &InvokeRequest,
+    ) -> anyhow::Result<InvokeResponse> {
+        self.print_request(req);
+        let work_dir = self.cfg.work_dir.join(req.id.to_hyphenated().to_string());
```

Comment:
Probably conflicts with src/invoker/src/executor.rs#L113

Reply:
That's the reason why duplicated code should be removed

Reply:
I don't think it is a serious bug.
Instead, request-id is mentioned in the path two times.

Of course, it's still an inconvenience and I fixed it, thanks.